app/src/main/java/uk/ac/ebi/atlas/experimentpage/cellplot/CellPlotService.java

```diff
@@ -103,27 +103,27 @@ public ImmutableSet<TSnePoint> expressionPlot(String experimentAccession,
     }
 
     public ImmutableMap<String, JsonObject> fetchDefaultPlotMethodWithParameterisation(String experimentAccession) {
-        var umap = "umap";
-        var tsne = "tsne";
+
         var defaultcellPlots = cellPlotDao.fetchDefaultPlotMethodWithParameterisation(experimentAccession);
 
         ImmutableMap.Builder<String, JsonObject> defaultPlotTypeAndOptions = new ImmutableMap.Builder<>();
-        if (!defaultcellPlots.isEmpty()) {
-            defaultPlotTypeAndOptions.put(umap, getMiddleElement(defaultcellPlots.get(umap)));
-            defaultPlotTypeAndOptions.put(tsne, getMiddleElement(defaultcellPlots.get(tsne)));
-        }
+
+        defaultcellPlots.forEach((method, options) -> defaultPlotTypeAndOptions.put(method, getMiddleElement(defaultcellPlots.get(method))));
+
         return defaultPlotTypeAndOptions.build();
     }
 
-    private static JsonObject getMiddleElement(List plotOptions) {
-
-        if (plotOptions.size() % 2 == 0) { // even number
-            Object umapEvenItem = plotOptions.get((plotOptions.size() / 2 - 1));
-            return (JsonObject) umapEvenItem;
-        } else { //odd number
-            Object umapOddItem = plotOptions.get((plotOptions.size() / 2));
-            return (JsonObject) umapOddItem;
-        }
+    private JsonObject getMiddleElement(List plotOptions) {
+        if (!plotOptions.isEmpty() && plotOptions.size() > 0) {
```

Comment:
We might also need a `null` check here. What do you think?

Reply:
Yes, as a safe side.